controller/api/destination/watcher/ip_watcher.go

```diff
@@ -285,10 +303,18 @@ func (iw *IPWatcher) getOrNewServiceSubscriptions(clusterIP string) *serviceSubs
 			pods := []*corev1.Pod{}
 			for _, obj := range objs {
 				if pod, ok := obj.(*corev1.Pod); ok {
-					// Skip pods with HostNetwork.
-					if !pod.Spec.HostNetwork {
-						pods = append(pods, pod)
+					// Skip pods with HostNetwork
+					if pod.Spec.HostNetwork {
+						continue
 					}
+
+					// Ignore terminated pods,
+					// their IPs can be reused for new Running pods
+					if podTerminated(pod) {
```

Comment:
Not 100% sure if this change is need. Looking forward to any feedback

Reply:
I think we want to omit terminated pods any time we are using the pod ip index.  So yes, I think this makes sense.